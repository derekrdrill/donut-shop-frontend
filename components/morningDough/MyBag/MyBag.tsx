import * as React from 'react';
import { useRouter } from 'next/router';
import {
  FormControl,
  FormControlLabel,
  FormLabel,
  Grid,
  IconButton,
  Radio,
  RadioGroup,
  Tooltip,
  Typography,
} from '@mui/material';
import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
import EditIcon from '@mui/icons-material/Edit';
import Info from '@mui/icons-material/Info';
import {
  ApplePay,
  Ach,
  CashAppPay,
  CreditCard,
  GooglePay,
  PaymentForm,
} from 'react-square-web-payments-sdk';

import GlobalContext from '../../../context/GlobalContext';

import MyBagItemButterText from './components/MyBagItemButterText';
import MyBagItemCreamCheeseText from './components/MyBagItemCreamCheeseText';
import MyBagItemDairyText from './components/MyBagItemDairyText';
import MyBagItemFlavorText from './components/MyBagItemFlavorText';
import MyBagItemIceText from './components/MyBagItemIceText';
import MyBagItemSugarText from './components/MyBagItemSugarText';

import { setModalItem } from '../../common/Modal/actions/ModalActions';
import { deleteFromMyBag } from './actions/MyBagActions';

import {
  MyBagCard,
  MyBagItemContainer,
  MyBagItemImage,
  MyBagItemRow,
  MyBagRoot,
  MyBagPaymentContainer,
} from './style';

const MyBag = () => {
  const router = useRouter();

  const {
    dispatch,
    state: { myBag },
  } = React.useContext(GlobalContext);

  const [paymentType, setPaymentType] = React.useState<string>('cc');

  React.useEffect(() => {
    if (myBag.length < 1) {
      router.push('/');
    }
  }, [myBag]);

  const subTotalNum = myBag?.reduce((n, { price }) => n + price, 0) ?? 0.0;
  const subTotalStr = `＄${subTotalNum?.toFixed(2)}`;
  const salesTaxNum = subTotalNum * 0.06;
  const salesTaxStr = `＄${salesTaxNum?.toFixed(2)}`;
  const total = `＄${(subTotalNum + salesTaxNum).toFixed(2)}`;

  return (
    <MyBagRoot container>
      <Grid item xs={12}>
        <Grid container justifyContent='center' spacing={4}>
          <Typography variant='h2'>Checkout</Typography>
        </Grid>
      </Grid>
      <Grid item xs={1} />
      <MyBagCard item xs={12} md={5} lg={6}>
        <Typography variant='h6'>My bag</Typography>
        <Grid item container>
          <MyBagItemContainer item xs={12}>
            {myBag.map(myBagItem => (
              <MyBagItemRow container key={myBagItem.orderID}>
                <Grid item xs={2}>
                  <Grid container justifyContent='center'>
                    <IconButton
                      onClick={
                        /* istanbul ignore next */
                        () =>
                          router.push(
                            `/menu/${myBagItem?.category ?? 'hotDrinks'}/${
                              myBagItem?.menuItemID ?? 'hotCoffee'
                            }/${myBagItem?.orderID}`,
                          )
                      }
                    >
                      <EditIcon color='info' />
                    </IconButton>
                    <IconButton
                      onClick={
                        /* istanbul ignore next */
                        () =>
                          dispatch(
                            setModalItem(
                              () => {
                                dispatch(deleteFromMyBag(myBagItem.orderID, myBag));
                                dispatch(setModalItem(null, false, null, null));
                              },
                              true,
                              <Typography variant='h6'>{`Are you sure you want to delete ${
                                myBagItem.quantity
                              } ${myBagItem.name}${
                                myBagItem.quantity > 1 ? `s` : ``
                              }?`}</Typography>,
                              'Delete item',
                              'error',
                              'Yes, Delete',
                            ),
                          )
                      }
                    >
                      <DeleteForeverIcon color='error' />
                    </IconButton>
                  </Grid>
                </Grid>
                <Grid item xs={1} />
                <Grid item xs={2}>
                  <MyBagItemImage src={myBagItem.menuItemImage} />
                </Grid>
                <Grid item xs={5}>
                  <Typography variant='subtitle1'>{myBagItem.name}</Typography>
                  <MyBagItemDairyText myBagItem={myBagItem} />
                  <MyBagItemFlavorText myBagItem={myBagItem} />
                  <MyBagItemIceText myBagItem={myBagItem} />
                  <MyBagItemSugarText myBagItem={myBagItem} />
                  <MyBagItemCreamCheeseText myBagItem={myBagItem} />
                  <MyBagItemButterText myBagItem={myBagItem} />
                </Grid>
                <Grid item xs={2}>
                  <Grid container>
                    <Grid item xs={12}>
                      <Typography variant='subtitle2'>x{myBagItem.quantity}</Typography>
                    </Grid>
                    <Grid item xs={12}>
                      <Typography variant='h6'>${myBagItem.price.toFixed(2)}</Typography>
                    </Grid>
                  </Grid>
                </Grid>
              </MyBagItemRow>
            ))}
          </MyBagItemContainer>
        </Grid>
        <Typography variant='subtitle1' textAlign='end'>{`Subtotal: ${subTotalStr}`}</Typography>
        <Typography variant='subtitle1' textAlign='end'>{`Sales tax: ${salesTaxStr}`}</Typography>
        <Typography variant='h5' textAlign='end' sx={{ paddingTop: 2 }}>
          {`Total: ${total}`}
        </Typography>
      </MyBagCard>
      <br />
      <MyBagCard item xs={12} md={5} lg={3}>
        <Typography variant='h6'>Payment details</Typography>
        <MyBagPaymentContainer>
          <Grid container>
            <Grid item xs={12}>
              <FormControl>
                <FormLabel>Select a payment type</FormLabel>
                <RadioGroup
                  value={paymentType}
                  onChange={e => setPaymentType(e.currentTarget.value)}
                >
                  <FormControlLabel
                    value='cc'
                    control={<Radio />}
                    label={
                      <Grid container justifyContent='space-between'>
                        <Typography variant='subtitle2'>Credit Card</Typography>
                        <Tooltip
                          placement='right'
                          title={
                            <>
                              <Typography fontWeight='bold' variant='subtitle2'>
                                Use the following <em>(from Square sandbox)</em>:
                              </Typography>
                              <br />
                              <Typography variant='subtitle2'>
                                <strong>Card number:</strong> 4111 1111 1111 1111
                              </Typography>
                              <Typography variant='subtitle2'>
                                <strong>Security code:</strong> 111
                              </Typography>
                              <Typography variant='subtitle2'>
                                <strong>Expiration:</strong> Any future date
                              </Typography>
                            </>
                          }
                        >
                          <Info sx={{ height: 17, paddingLeft: 1, width: 17 }} />
                        </Tooltip>
                      </Grid>
                    }
                  />
                  <FormControlLabel
                    value='ach'
                    control={<Radio />}
                    label={
                      <Grid container justifyContent='space-between'>
                        <Typography variant='subtitle2'>Bank Account</Typography>
                        <Tooltip
                          placement='right'
                          title={
                            <>
                              <Typography fontWeight='bold' variant='subtitle2'>
                                Use the following <em>(from Square sandbox)</em>:
                              </Typography>
                              <br />
                              <Typography variant='subtitle2'>
                                <strong>Bank user name:</strong> good_user
                              </Typography>
                              <Typography variant='subtitle2'>
                                <strong>Bank password:</strong> good_pass
                              </Typography>
                            </>
                          }
                        >
                          <Info sx={{ height: 17, paddingLeft: 1, width: 17 }} />
                        </Tooltip>
                      </Grid>
                    }
                  />
                  <FormControlLabel
                    value='google'
                    control={<Radio />}
                    label={
                      <Grid container justifyContent='space-between'>
                        <Typography variant='subtitle2'>Google Pay</Typography>
                        <Info sx={{ height: 17, paddingLeft: 1, width: 17 }} />
                      </Grid>
                    }
                  />
                  {/* <FormControlLabel value='apple' control={<Radio />} label='Apple Pay' /> */}
                  <FormControlLabel
                    value='cashApp'
                    control={<Radio />}
                    label={
                      <Grid container justifyContent='space-between'>
                        <Typography variant='subtitle2'>Cash App</Typography>
                        <Info sx={{ height: 17, paddingLeft: 1, width: 17 }} />
                      </Grid>
                    }
                  />
                </RadioGroup>
              </FormControl>
            </Grid>
          </Grid>
          <Grid container>
            <PaymentForm
              applicationId={process.env.NEXT_PUBLIC_SQUARE_APP_ID ?? ''}
              cardTokenizeResponseReceived={(token, verifiedBuyer) => {
                console.log('token:', token);
                console.log('verifiedBuyer:', verifiedBuyer);
              }}
              createPaymentRequest={() => ({
                countryCode: 'US',
                currencyCode: 'USD',
                total: {
                  amount: '1.00',
                  label: 'Total',
                },
              })}
              locationId={process.env.NEXT_PUBLIC_SQUARE_LOCATION_ID ?? ''}
            >
              <ApplePay style={{ display: paymentType === 'apple' ? 'inline-block' : 'none' }} />
              <GooglePay style={{ display: paymentType === 'google' ? 'inline-block' : 'none' }} />
              <CashAppPay
                style={{ display: paymentType === 'cashApp' ? 'inline-block' : 'none' }}
              />
              <Grid container sx={{ display: paymentType === 'cc' ? 'inline-block' : 'none' }}>
                <CreditCard />
              </Grid>
              <Grid container sx={{ display: paymentType === 'ach' ? 'inline-block' : 'none' }}>
                <Ach
                  accountHolderName='Guest User'
                  redirectURI='www.morning-dough.com'
                  transactionId='123'
                />
              </Grid>
            </PaymentForm>
          </Grid>
        </MyBagPaymentContainer>
      </MyBagCard>
    </MyBagRoot>
  );
};

export default MyBag;
