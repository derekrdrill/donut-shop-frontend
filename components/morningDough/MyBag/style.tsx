import styled from 'styled-components';
import { Badge, Grid } from '@mui/material';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const MyBagRoot = styled(Grid)({
  marginTop: 40,
});

export const MyBagIconSVG = styled(ShoppingBagIcon)({
  marginTop: 7,
  transition: 'all 250ms linear',
  transform: 'scale(1.4)',
});

export const MyBagIconBadge = styled(Badge)({
  '.MuiBadge-badge': {
    backgroundColor: 'white',
    color: '#6a017f',
    border: '1px #6a017f solid',
    transition: 'all 250ms linear',
  },
});

export const MyBagCard = styled(Grid)({
  border: '1px solid gainsboro',
  borderRadius: 5,
  margin: '0px 10px !important',
  padding: 20,
});

export const MyBagItemContainer = styled(Grid)({
  height: 390,
  overflowY: 'auto',
});

export const MyBagItemRow = styled(Grid)({
  borderBottom: '1px solid gainsboro',
  borderTop: '1px solid gainsboro',
  padding: 15,
});

export const MyBagItemImage = styled.img({
  height: 40,
  width: 40,
});

export const MyBagIconContainer = styled(Grid)({
  '&:hover': {
    '.MuiSvgIcon-root': {
      color: '#a501c6',
      transform: 'scale(1.6)',
    },
    '.MuiBadge-root': {
      transform: 'scale(1.2)',
      zIndex: 4,
    },
  },
  cursor: 'pointer',
});

export const MyBagPaymentContainer = styled(Grid)({
  borderTop: '1px solid gainsboro',
  padding: 15,
});
