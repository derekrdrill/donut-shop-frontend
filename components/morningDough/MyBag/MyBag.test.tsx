jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

import GlobalContext from '../../context/GlobalContext';

import MyBag from './MyBag';

describe('MyBag unit tests', () => {
  const myBag = [
    {
      bottled: false,
      butter: null,
      category: 'donuts',
      creamCheese: null,
      dairy: null,
      flavor: null,
      ice: 'normal',
      menuItemID: 'chocolateDonut',
      menuItemImage: 'http://localhost:3000/chocolateDonut.png',
      name: 'Chocolate',
      orderID: 'chocolateDonut-3218737128',
      quantity: 5,
      size: 'md',
      soda: false,
      subCategory: 'donuts',
      sugar: false,
    },
    {
      bottled: false,
      butter: null,
      category: 'drink',
      creamCheese: null,
      dairy: 'cream',
      flavor: 'frenchVanilla',
      ice: 'normal',
      menuItemID: 'hotCoffee',
      menuItemImage: 'http://localhost:3000/hotCoffee.png',
      name: 'Hot Coffee',
      orderID: '1681765872483-menuItemID',
      quantity: 3,
      size: 'md',
      soda: false,
      subCategory: 'hotDrinks',
      sugar: true,
    },
  ];

  it('renders as expected', () => {
    const result = renderer
      .create(
        <GlobalContext.Provider value={{ dispatch: jest.fn(), state: { myBag } }}>
          <MyBag />
        </GlobalContext.Provider>,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
