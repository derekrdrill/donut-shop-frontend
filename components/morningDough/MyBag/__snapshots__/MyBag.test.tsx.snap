// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBag unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root MuiGrid-container sc-beqWaB cwMraW css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-1xnaaoc-MuiGrid-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
      >
        Checkout
      </h2>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
  />
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-md-5 sc-hLseeU hWZUsq css-doz332-MuiGrid-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      My bag
    </h6>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 sc-eDDNvR crpmXz css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-container sc-jTrPJq bkbSJV css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          />
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <img
              className="sc-gLDzan bOZWjX"
              src="http://localhost:3000/chocolateDonut.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
            >
              Chocolate
            </h6>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                x
                5
              </h6>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
                  data-testid="EditIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </button>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
                  data-testid="DeleteForeverIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          />
        </div>
        <div
          className="MuiGrid-root MuiGrid-container sc-jTrPJq bkbSJV css-11lq3yg-MuiGrid-root"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          />
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <img
              className="sc-gLDzan bOZWjX"
              src="http://localhost:3000/hotCoffee.png"
            />
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
            >
              Hot Coffee
            </h6>
            <p
              className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              data-testid="dairy-text"
            >
              CREAM
            </p>
            <p
              className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              data-testid="flavor-text"
            >
              FRENCH VANILLA
            </p>
            <p
              className="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
              data-testid="sugar-text"
            >
              SUGAR
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <h6
                className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              >
                x
                3
              </h6>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <div
              className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
            >
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
                  data-testid="EditIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                  />
                </svg>
              </button>
              <button
                className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
                  data-testid="DeleteForeverIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12 1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 css-u78q51-MuiGrid-root"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 MuiGrid-grid-sm-5 sc-hLseeU hWZUsq css-pmp550-MuiGrid-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Payment details
    </h6>
    <div
      className="MuiGrid-root sc-fsQiph dCoEro css-vj1n65-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;
